crates/tinymist-query/src/upstream/complete/ext.rs

```diff
@@ -257,17 +260,25 @@ impl CompletionContext<'_> {
                 // range: Some(range),
                 ..Default::default()
             };
+            let base_item = CompletionItem {
+                kind: Some(CompletionItemKind::SNIPPET),
+                label: snippet.label.clone().into(),
+                detail: Some(snippet.description.clone().into()),
+                insert_text_format: Some(InsertTextFormat::SNIPPET),
+                ..Default::default()
+            };
             if let Some(node_before_before_cursor) = &node_before_before_cursor {
                 let node_content = node.get().clone().into_text();
                 let before = TextEdit {
                     range: self.ctx.to_lsp_range(rng.start..self.from, &src),
```

Comment:
The range of main edit can only start at the cursor, but `rng.start` is before the cursor.

Reply:
Probably not, the lsp spec says, "it must be a [single line] and they must contain the position at which completion has been requested," rather than "start with the position".

Reply:
Yes, that's why it is ended in self.from.

Reply:
Maybe I should use the insertandreplace capability of lsp

Reply:
As the lsp spec said above, `rng.start..self.from` may span across multiple lines but the spec requires it MUST be in a single line so I don't think this is a valid fix. Btw, I checked it in my local neovim, and at least the case in #986 can work as expected. It might be worth to checking whether some neovim completion plugins don't work while others do work. I use the default LazyVim plus the `tinymist.lua` from [the commit](https://github.com/Myriad-Dreamin/tinymist/pull/993).